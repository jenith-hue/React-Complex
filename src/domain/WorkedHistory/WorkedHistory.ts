import { RACCOLOR } from '@rentacenter/racstrap';

export enum WorkedHistoryStatusLegend {
  LeftMessage = 'LeftMessage',
  Commitment = 'Commitment',
  NoContact = 'NoContact',
  BrokenCommitment = 'BrokenCommitment',
  TextSent = 'TextSent',
  TextReceived = 'TextReceived',
}

export interface WorkedHistoryStatusLegendEntityProps {
  label: string;
  color?: string;
}

export const WorkedHistoryStatusLegendNames: Partial<
  Record<WorkedHistoryStatusLegend, WorkedHistoryStatusLegendEntityProps>
> = {
  [WorkedHistoryStatusLegend.LeftMessage]: {
    label: 'Left Message',
    color: RACCOLOR.TUFTS_BLUE,
  },
  [WorkedHistoryStatusLegend.Commitment]: {
    label: 'Commitment',
    color: RACCOLOR.HONEYCOMB,
  },
  [WorkedHistoryStatusLegend.NoContact]: {
    label: 'No Contact',
    color: RACCOLOR.RICH_LILAC,
  },
  [WorkedHistoryStatusLegend.BrokenCommitment]: {
    label: 'Broken Commitment',
    color: RACCOLOR.PUNCH,
  },
  [WorkedHistoryStatusLegend.TextSent]: {
    label: 'Text Sent',
    color: RACCOLOR.BLUE_MANA,
  },
  [WorkedHistoryStatusLegend.TextReceived]: {
    label: 'Text Received',
    color: RACCOLOR.INTERGALACTIC_COWBOY,
  },
};

export interface LogWorkedHistoryPayload {
  storeNumber: string;
  activityDate: string;
  notes: string;
  callResultRefCode: string;
  commitmentId: string;
  acctActivityRefCode: string;
  daysPastDue: string;
  customerId: string | number;
  phoneNumberDialed: string;
  coWorkerId: string;
  receiptId: string;
  callTime: string;
}
