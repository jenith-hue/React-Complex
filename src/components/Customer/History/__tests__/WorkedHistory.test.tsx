/* eslint-disable sonarjs/no-duplicate-string */
/* eslint-disable react/display-name */
/* eslint-disable sonarjs/no-identical-functions */

import React from 'react';
import renderer from 'react-test-renderer';
import RouteData from 'react-router-dom';

import { WorkedHistory } from '../WorkedHistory';
import * as Utils from '../../../../utils/utils';

interface PropWithChildren {
  children: JSX.Element;
}

interface MockRACTableProps {
  renderTableHead: () => JSX.Element;
  renderTableContent: () => JSX.Element;
}

jest.mock(
  'react-infinite-scroll-component',
  () =>
    ({ children, ...rest }: PropWithChildren) =>
      (
        <div data-testid="InfiniteScroll" {...rest}>
          {children}
        </div>
      )
);

jest.mock('@rentacenter/racstrap', () => ({
  ...jest.requireActual('@rentacenter/racstrap'),
  Typography: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="Typography" {...rest}>
      {children}
    </div>
  ),
  RACButton: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="RACButton" {...rest}>
      {children}
    </div>
  ),
  RACCard: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="RACCard" {...rest}>
      {children}
    </div>
  ),
  RACTableCell: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="RACTableCell" {...rest}>
      {children}
    </div>
  ),
  RACTableRow: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="RACTableRow" {...rest}>
      {children}
    </div>
  ),
  RACRowTip: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="RACRowTip" {...rest}>
      {children}
    </div>
  ),
  RACTable: ({ renderTableHead, renderTableContent }: MockRACTableProps) => (
    <div data-testid="RACTable">
      <div id="table-head">{renderTableHead()}</div>
      <div id="table-content">{renderTableContent()}</div>
    </div>
  ),
  makeStyles: () => () => ({}),
}));

jest.mock('../../../../context/WorkedHistory/WorkedHistoryProvider', () => ({
  useWorkedHistoryActions: () => ({
    fetchWorkedHistory: jest.fn(),
  }),
  useWorkedHistory: () => ({
    workedHistory: [
      {
        workedDate: '2014-05-31T00:04:21.0-0500',
        notes: 'Test note1',
        accountActivityType: {
          code: 'CALLC',
          description: 'Text Sent To Phone',
        },
        accountActivitySubType: {
          code: '2WAYTXT',
          description: '2 WAY Text',
        },
        callResultType: {
          code: 'NA',
          description: 'No Answer',
        },
        commitment: {
          commitmentDate: '2014-05-31T00:04:21.0-0500',
          commitmentType: {
            code: 'PY',
            description: 'Payment',
          },
          commitmentStatus: {
            code: 'GD',
            description: 'Good',
          },
        },
        dueDate: '2021-07-25',
        daysPastDue: '10',
        phoneNumberDialed: '214-726-0111',
        relation: 'Friend',
        coWorker: {
          id: '236268',
          firstName: 'Tim',
          lastName: 'Sutton',
        },
      },
      {
        workedDate: '2014-05-31T00:04:21.0-0500',
        notes: 'Test note2',
        accountActivityType: {
          code: 'CALLC',
          description: 'Text Sent To Phone',
        },
        accountActivitySubType: {
          code: '2WAYTXT',
          description: '2 WAY Text',
        },
        callResultType: {
          code: 'LMAL',
          description: 'Text Sent To Phone',
        },
        commitment: {
          commitmentDate: '2014-05-31T00:04:21.0-0500',
          commitmentType: {
            code: 'PY',
            description: 'Payment',
          },
          commitmentStatus: {
            code: 'GD',
            description: 'Good',
          },
        },
        dueDate: '2021-07-25',
        daysPastDue: '10',
        phoneNumberDialed: '214-726-0111',
        relation: 'Friend',
        coWorker: {
          id: '236268',
          firstName: 'Tim',
          lastName: 'Sutton',
        },
      },
      {
        workedDate: '2014-05-31T00:04:21.0-0500',
        notes: 'Test note3',
        accountActivityType: {
          code: 'CALLC',
          description: 'Text Sent To Phone',
        },
        accountActivitySubType: {
          code: '2WAYTXT',
          description: '2 WAY Text',
        },
        callResultType: {
          code: 'NOT_EXISTING_CODE',
          description: 'Text Sent To Phone',
        },
        commitment: {
          commitmentDate: '2014-05-31T00:04:21.0-0500',
          commitmentType: {
            code: 'PY',
            description: 'Payment',
          },
          commitmentStatus: {
            code: 'GD',
            description: 'Good',
          },
        },
        dueDate: '2021-07-25',
        daysPastDue: '10',
        phoneNumberDialed: '214-726-0111',
        relation: 'Friend',
        coWorker: {
          id: '236268',
          firstName: 'Tim',
          lastName: 'Sutton',
        },
      },
      {
        workedDate: '2014-05-31T00:04:21.0-0500',
        notes: 'Test note 4',
        accountActivityType: {
          code: 'CMT',
          description: 'Text Sent To Phone',
        },
        accountActivitySubType: {
          code: '2WAYTXT',
          description: '2 WAY Text',
        },
        callResultType: {
          code: 'LMAL',
          description: 'Text Sent To Phone',
        },
        commitment: {
          commitmentDate: '2014-05-31T00:04:21.0-0500',
          commitmentType: {
            code: 'PY',
            description: 'Payment',
          },
          commitmentStatus: {
            code: 'BR',
            description: 'Broken',
          },
        },
        dueDate: '2021-07-25',
        daysPastDue: '10',
        phoneNumberDialed: '214-726-0111',
        relation: 'Friend',
        coWorker: {
          id: '236268',
          firstName: 'Tim',
          lastName: 'Sutton',
        },
      },
      {
        workedDate: '2014-05-31T00:04:21.0-0500',
        notes: 'Test note 5',
        accountActivityType: {
          code: 'CMT',
          description: 'Text Sent To Phone',
        },
        accountActivitySubType: {
          code: '2WAYTXT',
          description: '2 WAY Text',
        },
        callResultType: {
          code: 'LMAL',
          description: 'Text Sent To Phone',
        },
        commitment: {
          commitmentDate: '2014-05-31T00:04:21.0-0500',
          commitmentType: {
            code: 'PY',
            description: 'Payment',
          },
          commitmentStatus: {
            code: 'NOT_EXISTING_CODE',
            description: 'Not existing, default to white',
          },
        },
        dueDate: '2021-07-25',
        daysPastDue: '10',
        phoneNumberDialed: '214-726-0111',
        relation: 'Friend',
        coWorker: {
          id: '236268',
          firstName: 'Tim',
          lastName: 'Sutton',
        },
      },
      {
        workedDate: '2014-05-31T00:04:21.0-0500',
        notes: 'Test note 6',
        accountActivityType: {
          code: 'TXTS',
          description: 'Text Sent To Phone',
        },
        accountActivitySubType: {
          code: '2WAYTXT',
          description: '2 WAY Text',
        },
        callResultType: {
          code: 'LMAL',
          description: 'Text Sent To Phone',
        },
        commitment: {
          commitmentDate: '2014-05-31T00:04:21.0-0500',
          commitmentType: {
            code: 'PY',
            description: 'Payment',
          },
          commitmentStatus: {
            code: 'NOT_EXISTING_CODE',
            description: 'Not existing, default to white',
          },
        },
        dueDate: '2021-07-25',
        daysPastDue: '10',
        phoneNumberDialed: '214-726-0111',
        relation: 'Friend',
        coWorker: {
          id: '236268',
          firstName: 'Tim',
          lastName: 'Sutton',
        },
      },
      {
        workedDate: '2014-05-31T00:04:21.0-0500',
        notes: 'Test note 7',
        accountActivityType: {
          code: 'TXTR',
          description: 'Text Sent To Phone',
        },
        accountActivitySubType: {
          code: '2WAYTXT',
          description: '2 WAY Text',
        },
        callResultType: {
          code: 'LMAL',
          description: 'Text Sent To Phone',
        },
        commitment: {
          commitmentDate: '2014-05-31T00:04:21.0-0500',
          commitmentType: {
            code: 'PY',
            description: 'Payment',
          },
          commitmentStatus: {
            code: 'GD',
            description: 'Good',
          },
        },
        dueDate: '2021-07-25',
        daysPastDue: '10',
        phoneNumberDialed: '214-726-0111',
        relation: 'Friend',
        coWorker: {
          id: '236268',
          firstName: 'Tim',
          lastName: 'Sutton',
        },
      },
    ],
  }),
}));

// Notice that the below is required, even if there is an another
// jest.spyOn in the test.
// Otherwise, we'll get the following: TypeError: Cannot redefine property: useLocation
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: '',
  }),
}));
describe('WorkedHistory', () => {
  it('renders correctly', () => {
    jest.spyOn(RouteData, 'useLocation').mockImplementationOnce(() => ({
      pathname: '/testroute',
      search: '',
      hash: '',
      state: {
        customer: {
          customerFirstName: 'First',
          customerLastName: 'Last',
          customerId: '123ID',
        },
      },
    }));
    jest
      .spyOn(Utils, 'formatDateString')
      .mockImplementation(() => '12/12/2021');
    jest
      .spyOn(Utils, 'formatStringDateHoursAndMinutes')
      .mockImplementation(() => '11 AM');
    const tree = renderer.create(<WorkedHistory />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
