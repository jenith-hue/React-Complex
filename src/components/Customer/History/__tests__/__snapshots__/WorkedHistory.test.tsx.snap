// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkedHistory renders correctly 1`] = `
Array [
  <div>
    <div
      data-testid="legendItemListTestId"
    >
      <div
        className=""
        data-testid="legendItemTestId"
        onClick={[Function]}
      >
        <span />
        <span>
          Left Message
        </span>
      </div>
      <div
        className=""
        data-testid="legendItemTestId"
        onClick={[Function]}
      >
        <span />
        <span>
          Commitment
        </span>
      </div>
      <div
        className=""
        data-testid="legendItemTestId"
        onClick={[Function]}
      >
        <span />
        <span>
          No Contact
        </span>
      </div>
      <div
        className=""
        data-testid="legendItemTestId"
        onClick={[Function]}
      >
        <span />
        <span>
          Broken Commitment
        </span>
      </div>
      <div
        className=""
        data-testid="legendItemTestId"
        onClick={[Function]}
      >
        <span />
        <span>
          Text Sent
        </span>
      </div>
      <div
        className=""
        data-testid="legendItemTestId"
        onClick={[Function]}
      >
        <span />
        <span>
          Text Received
        </span>
      </div>
    </div>
  </div>,
  <div
    data-testid="InfiniteScroll"
    dataLength={7}
    height={330}
    loader={null}
    next={[Function]}
  >
    <div
      data-testid="RACTable"
    >
      <div
        id="table-head"
      >
        <div
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Worked Date & Time
          </div>
        </div>
        <div
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Worked Result
          </div>
        </div>
        <div
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Notes
          </div>
        </div>
        <div
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Commitment Status
          </div>
        </div>
        <div
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Due Date
          </div>
        </div>
        <div
          align="center"
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Days Late
          </div>
        </div>
        <div
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Phone # Dialed
          </div>
        </div>
        <div
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Relation
          </div>
        </div>
        <div
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Source
          </div>
        </div>
      </div>
      <div
        id="table-content"
      >
        <div
          data-testid="RACTableRow"
          hover={true}
        >
          <div
            classes={
              Object {
                "root": undefined,
              }
            }
            data-testid="RACTableCell"
          >
            <div>
              <div
                classes={
                  Object {
                    "root": undefined,
                  }
                }
                data-testid="RACRowTip"
                tipColor="#ffffff"
              />
              <div
                data-testid="Typography"
                display="inline"
                variant="body2"
              >
                12/12/2021 11 AM
              </div>
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              No Answer
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              noWrap={true}
              variant="body2"
            >
              <div
                color="primary"
                data-testid="Typography"
                noWrap={true}
                variant="body2"
              >
                <span
                  onClick={[Function]}
                >
                  Test note1
                </span>
              </div>
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              Good
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              12/12/2021
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              align="center"
              data-testid="Typography"
              variant="body2"
            />
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              (***) ***-****
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              Tim Sutton
            </div>
          </div>
        </div>
        <div
          data-testid="RACTableRow"
          hover={true}
        >
          <div
            classes={
              Object {
                "root": undefined,
              }
            }
            data-testid="RACTableCell"
          >
            <div>
              <div
                classes={
                  Object {
                    "root": undefined,
                  }
                }
                data-testid="RACRowTip"
                tipColor="#ffffff"
              />
              <div
                data-testid="Typography"
                display="inline"
                variant="body2"
              >
                12/12/2021 11 AM
              </div>
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              Text Sent To Phone
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              noWrap={true}
              variant="body2"
            >
              <div
                color="primary"
                data-testid="Typography"
                noWrap={true}
                variant="body2"
              >
                <span
                  onClick={[Function]}
                >
                  Test note2
                </span>
              </div>
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              Good
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              12/12/2021
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              align="center"
              data-testid="Typography"
              variant="body2"
            />
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              (***) ***-****
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              Tim Sutton
            </div>
          </div>
        </div>
        <div
          data-testid="RACTableRow"
          hover={true}
        >
          <div
            classes={
              Object {
                "root": undefined,
              }
            }
            data-testid="RACTableCell"
          >
            <div>
              <div
                classes={
                  Object {
                    "root": undefined,
                  }
                }
                data-testid="RACRowTip"
                tipColor="#ffffff"
              />
              <div
                data-testid="Typography"
                display="inline"
                variant="body2"
              >
                12/12/2021 11 AM
              </div>
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              Text Sent To Phone
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              noWrap={true}
              variant="body2"
            >
              <div
                color="primary"
                data-testid="Typography"
                noWrap={true}
                variant="body2"
              >
                <span
                  onClick={[Function]}
                >
                  Test note3
                </span>
              </div>
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              Good
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              12/12/2021
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              align="center"
              data-testid="Typography"
              variant="body2"
            />
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              (***) ***-****
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              Tim Sutton
            </div>
          </div>
        </div>
        <div
          data-testid="RACTableRow"
          hover={true}
        >
          <div
            classes={
              Object {
                "root": undefined,
              }
            }
            data-testid="RACTableCell"
          >
            <div>
              <div
                classes={
                  Object {
                    "root": undefined,
                  }
                }
                data-testid="RACRowTip"
                tipColor="#ffffff"
              />
              <div
                data-testid="Typography"
                display="inline"
                variant="body2"
              >
                12/12/2021 11 AM
              </div>
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              Text Sent To Phone
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              noWrap={true}
              variant="body2"
            >
              <div
                color="primary"
                data-testid="Typography"
                noWrap={true}
                variant="body2"
              >
                <span
                  onClick={[Function]}
                >
                  Test note 4
                </span>
              </div>
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              Broken
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              12/12/2021
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              align="center"
              data-testid="Typography"
              variant="body2"
            />
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              (***) ***-****
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              Tim Sutton
            </div>
          </div>
        </div>
        <div
          data-testid="RACTableRow"
          hover={true}
        >
          <div
            classes={
              Object {
                "root": undefined,
              }
            }
            data-testid="RACTableCell"
          >
            <div>
              <div
                classes={
                  Object {
                    "root": undefined,
                  }
                }
                data-testid="RACRowTip"
                tipColor="#ffffff"
              />
              <div
                data-testid="Typography"
                display="inline"
                variant="body2"
              >
                12/12/2021 11 AM
              </div>
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              Text Sent To Phone
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              noWrap={true}
              variant="body2"
            >
              <div
                color="primary"
                data-testid="Typography"
                noWrap={true}
                variant="body2"
              >
                <span
                  onClick={[Function]}
                >
                  Test note 5
                </span>
              </div>
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              Not existing, default to white
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              12/12/2021
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              align="center"
              data-testid="Typography"
              variant="body2"
            />
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              (***) ***-****
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              Tim Sutton
            </div>
          </div>
        </div>
        <div
          data-testid="RACTableRow"
          hover={true}
        >
          <div
            classes={
              Object {
                "root": undefined,
              }
            }
            data-testid="RACTableCell"
          >
            <div>
              <div
                classes={
                  Object {
                    "root": undefined,
                  }
                }
                data-testid="RACRowTip"
                tipColor="#ffffff"
              />
              <div
                data-testid="Typography"
                display="inline"
                variant="body2"
              >
                12/12/2021 11 AM
              </div>
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              Text Sent To Phone
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              noWrap={true}
              variant="body2"
            >
              <div
                color="primary"
                data-testid="Typography"
                noWrap={true}
                variant="body2"
              >
                <span
                  onClick={[Function]}
                >
                  Test note 6
                </span>
              </div>
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              Not existing, default to white
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              12/12/2021
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              align="center"
              data-testid="Typography"
              variant="body2"
            />
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              (***) ***-****
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              Tim Sutton
            </div>
          </div>
        </div>
        <div
          data-testid="RACTableRow"
          hover={true}
        >
          <div
            classes={
              Object {
                "root": undefined,
              }
            }
            data-testid="RACTableCell"
          >
            <div>
              <div
                classes={
                  Object {
                    "root": undefined,
                  }
                }
                data-testid="RACRowTip"
                tipColor="#ffffff"
              />
              <div
                data-testid="Typography"
                display="inline"
                variant="body2"
              >
                12/12/2021 11 AM
              </div>
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              Text Sent To Phone
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              noWrap={true}
              variant="body2"
            >
              <div
                color="primary"
                data-testid="Typography"
                noWrap={true}
                variant="body2"
              >
                <span
                  onClick={[Function]}
                >
                  Test note 7
                </span>
              </div>
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              Good
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              12/12/2021
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              align="center"
              data-testid="Typography"
              variant="body2"
            />
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              (***) ***-****
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              data-testid="Typography"
              variant="body2"
            >
              undefined undefined
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
